### ISSUE TITLE
fish_tests.cpp "Testing autosuggestion combining" is flakey

### ISSUE THREAD

**Issue by krader1961** [fish-shell/fish-shell#3471]
Starting a couple of days ago I started seeing the fourth test environment (the one that simulates OS X/macOS) fail with this output in the log:

```
Testing autosuggestion combining
line 2165: autosuggest_suggest_special() failed for command cd ~haha
Error: Test failed on line 2045: !comps.empty()
```

I've seen it take two restarts for the test to complete. It's unclear if this is a problem with the Travis test environment or represents a race condition in fish. It's hard to see how this could be due to the Travis test environment. Which is why I'm creating this issue. I'm going to immediately close it but wanted to make a record of this seemingly new test failure mode.

**Comment by krader1961** [comment]
I'm reopening this issue because it is happening frequently enough to be an impediment to further development. TBD is whether this is a bug in fish or Travis CI. Developer @floam believes he saw this failure mode as early as 2016-10-03. This failure caught my attention approximately three days ago (i.e., 2016-10-14).

Making the following trivial change causes the test to fail in all Travis fish environments rather than just the OS X one:

```
-    perform_one_autosuggestion_cd_test(L"cd ~haha", vars, L"ha/path1/path2/", __LINE__);
+    perform_one_autosuggestion_cd_test(L"cd ~arglebargle", vars, L"ha/path1/path2/", __LINE__);
```

So that tells me the issue isn't just that there might be a user account named "haha" on some of the Travis systems. It also hints that the problem isn't due to something in the Travis CI environment but is rather a bug we recently introduced to fish.

**Comment by floam** [comment]
As @krader1961 noted, I noticed this on OS X builds on October 3rd. 

This was announced October 4th:
https://blog.travis-ci.com/2016-10-04-osx-73-default-image-live/

**Comment by krader1961** [comment]
I give up. I ran the failing travis test environment 32 times. I saw just two failures; one each on different, very recent, commits. Those were the first two runs I executed. The next 30 runs, using different commits from the past week all passed. Including six runs for one of the aforementioned two commits that failed. I've also been continuously running the test on may macOS server without failure for the past two days. I also haven't seen any test failures since [Travis job 3161](https://travis-ci.org/fish-shell/fish-shell/builds/168495728) a day ago.

So it is hard to believe this is a problem with fish. More likely it is a problem in the Travis OS X environment that has possibly been fixed in the past day. I'm going to close this again as "can't reproduce".
